test/AutoDiff/sil_differentiability_witness.sil

```diff
@@ -78,3 +76,38 @@ sil_differentiability_witness hidden [parameters 0 1] [results 0] [where T : _Di
 // CHECK:   jvp: @AD__generic__jvp_src_0_wrt_0_1 : $@convention(thin) <τ_0_0 where τ_0_0 : _Differentiable> (@in_guaranteed τ_0_0, Float) -> (@out τ_0_0, @owned @callee_guaranteed (@in_guaranteed τ_0_0.TangentVector, Float) -> @out τ_0_0.TangentVector)
 // CHECK:   vjp: @AD__generic__vjp_src_0_wrt_0_1 : $@convention(thin) <τ_0_0 where τ_0_0 : _Differentiable> (@in_guaranteed τ_0_0, Float) -> (@out τ_0_0, @owned @callee_guaranteed (@in_guaranteed τ_0_0.TangentVector) -> (@out τ_0_0.TangentVector, Float))
 // CHECK: }
+
+// Test SIL differentiability witness for external function.
+
+sil @externalFn1 : $@convention(thin) (Float) -> Float
+
+sil @AD__externalFn1__jvp_src_0_wrt_0 : $@convention(thin) (Float) -> (Float, @owned @callee_guaranteed (Float) -> Float) {
```

Comment:
I wonder if we want to support "differentiability witness definition (has body) referring to JVP/VJP declarations (bodiless)"?

I wonder if the test still passes if `@AD__externalFn1__jvp_src_0_wrt_0` and `@AD__externalFn1__vjp_src_0_wrt_0` are made bodiless?

Reply:
Yes it passes. I'll add a test that does that. (Bodiless original with defined jvp/vjp seems like the most common case that will happen in practice so I want to keep a test that does it this way rather than modifying this test to have bodiless jvp/vjp).